### PR TITLE
Refactor messages regarding chat 

### PR DESCRIPTION
**Pull request by BeeHiveJava**

### CODE REVIEW

File: Soccar [Game-Server]/src/nl/soccar/gameserver/model/command/CommandListener.java [hunk 1]
```diff
@@ -23,7 +24,7 @@ public CommandListener(Privilege minimumPrivilege, boolean hostOnly) {
         this.hostOnly = hostOnly;
     }
 
-    public abstract void execute(PlayerWrapper player, RoomWrapper room, Command command);
+    public abstract void execute(PlayerWrapper player, Chat chat, Command command);
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/model/command/CommandListener.java:27]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/model/command/CommandListener.java [hunk 2]
```diff
@@ -23,7 +24,7 @@ public CommandListener(Privilege minimumPrivilege, boolean hostOnly) {
...
 
     public final void executePrivileged(PlayerWrapper player, Command command) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/model/command/CommandListener.java:29]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/model/Chat.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            return;
+        }
+
+        ChatMessage m = new ChatMessage(player.getPlayerId(), player.getPrivilege(), message);
+        room.getPlayers().stream()
+                .map(PlayerWrapper::getConnection)
+                .forEach(c -> c.send(m));
+    }
+
+    public void mutePlayer(PlayerWrapper issuer, String username) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/model/Chat.java:50]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/model/Chat.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+        Optional<PlayerWrapper> optional = room.getPlayers().stream().filter(p -> p.getUsername().equalsIgnoreCase(username)).findFirst();
+        if (!optional.isPresent()) {
+            return;
+        }
+
+        optional.get().getConnection().send(new ChatMessage(-1, Privilege.ADMINISTRATOR, "You have been muted."));
+    }
+
+    public void unmutePlayer(PlayerWrapper issuer, String username) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/model/Chat.java:67]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/model/Chat.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+/**
+ * @author Lesley
+ */
+public final class Chat {
+
+    private final List<String> mutedNames = new ArrayList<>();
+
+    private final RoomWrapper room;
+
+    public Chat(RoomWrapper room) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/model/Chat.java:22]: MINOR Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/model/Chat.java [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+        this.room = room;
+    }
+
+    public void sendChatMessage(PlayerWrapper player, String message) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/model/Chat.java:26]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/RoomWrapper.java [hunk 7]
```diff
@@ -29,14 +30,16 @@
...
     private final Map<PlayerWrapper, Team> teams = new HashMap<>();
 
-    private final List<String> mutedNames = new ArrayList<>();
-
     private final SessionWrapper session;
     private final Room room;
 
+    private final Chat chat;
+
     public RoomWrapper(SessionWrapper session, Room room) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/RoomWrapper.java:38]: MINOR Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/RoomWrapper.java [hunk 8]
```diff
@@ -29,14 +30,16 @@
...
         this.session = session;
         this.room = room;
+
+        chat = new Chat(this);
     }
 
     public void join(PlayerWrapper player, String password) {
```
Comment (ghost) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/RoomWrapper.java:45]: MINOR Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

**Comment by ghost** [comment]
SonarQube analysis reported 15 issues
* MINOR 15 minor

Watch the comments in this conversation to review them.

#### 7 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [CommandListener.java#L18](https://github.com/PTS3-S34A/Game-Server/blob/dde0d5a179ebe8082d9a1d312b5d23274b844a05/Soccar [Game-Server]/src/nl/soccar/gameserver/model/command/CommandListener.java#L18): Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [CommandListener.java#L22](https://github.com/PTS3-S34A/Game-Server/blob/dde0d5a179ebe8082d9a1d312b5d23274b844a05/Soccar [Game-Server]/src/nl/soccar/gameserver/model/command/CommandListener.java#L22): Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [BroadcastCommandListener.java#L15](https://github.com/PTS3-S34A/Game-Server/blob/dde0d5a179ebe8082d9a1d312b5d23274b844a05/Soccar [Game-Server]/src/nl/soccar/gameserver/model/command/listener/BroadcastCommandListener.java#L15): Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [MuteCommandListener.java#L14](https://github.com/PTS3-S34A/Game-Server/blob/dde0d5a179ebe8082d9a1d312b5d23274b844a05/Soccar [Game-Server]/src/nl/soccar/gameserver/model/command/listener/MuteCommandListener.java#L14): Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [UnmuteCommandListener.java#L14](https://github.com/PTS3-S34A/Game-Server/blob/dde0d5a179ebe8082d9a1d312b5d23274b844a05/Soccar [Game-Server]/src/nl/soccar/gameserver/model/command/listener/UnmuteCommandListener.java#L14): Document this public constructor. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [RoomWrapper.java#L82](https://github.com/PTS3-S34A/Game-Server/blob/dde0d5a179ebe8082d9a1d312b5d23274b844a05/Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/RoomWrapper.java#L82): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)
1. MINOR [RoomWrapper.java#L116](https://github.com/PTS3-S34A/Game-Server/blob/dde0d5a179ebe8082d9a1d312b5d23274b844a05/Soccar [Game-Server]/src/nl/soccar/gameserver/model/session/RoomWrapper.java#L116): Document this public method. [rule](http://194.135.91.67:9000/coding_rules#rule_key=squid%3AUndocumentedApi)

File: Soccar [Game-Server]/src/nl/soccar/gameserver/model/Chat.java [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+            String[] arguments = new String[components.length - 1];
+            System.arraycopy(components, 1, arguments, 0, arguments.length);
+
+            Command command = new Command(name, arguments);
+            CommandDispatcher.dispatch(player, command);
+            return;
+        }
+
+        if (mutedNames.contains(player.getUsername().toLowerCase())) {
+            player.getConnection().send(new ChatMessage(-1, Privilege.ADMINISTRATOR, "You have been muted and, therefor, cannot talk."));
```
Comment (Ranray) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/model/Chat.java:40]: Graag een constante toevoegen voor de -1 waarde in de ChatMessage.

File: Soccar [Game-Server]/src/nl/soccar/gameserver/model/Chat.java [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+
+        mutedNames.add(lower);
+        issuer.getConnection().send(new ChatMessage(-1, Privilege.ADMINISTRATOR, String.format("You have muted %s.", username)));
+
+        Optional<PlayerWrapper> optional = room.getPlayers().stream().filter(p -> p.getUsername().equalsIgnoreCase(username)).findFirst();
+        if (!optional.isPresent()) {
+            return;
+        }
+
+        optional.get().getConnection().send(new ChatMessage(-1, Privilege.ADMINISTRATOR, "You have been muted."));
```
Comment (Ranray) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/model/Chat.java:64]: Graag een constante toevoegen voor de -1 waarde in de ChatMessage.

File: Soccar [Game-Server]/src/nl/soccar/gameserver/model/Chat.java [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+    public void unmutePlayer(PlayerWrapper issuer, String username) {
+        mutedNames.remove(username.toLowerCase());
+        issuer.getConnection().send(new ChatMessage(-1, Privilege.ADMINISTRATOR, String.format("You have unmuted %s.", username)));
+
+        Optional<PlayerWrapper> optional = room.getPlayers().stream().filter(p -> p.getUsername().equalsIgnoreCase(username)).findFirst();
+        if (!optional.isPresent()) {
+            return;
+        }
+
+        optional.get().getConnection().send(new ChatMessage(-1, Privilege.ADMINISTRATOR, "You have been unmuted."));
```
Comment (Ranray) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/model/Chat.java:76]: Graag een constante toevoegen voor de -1 waarde in de ChatMessage.

File: Soccar [Game-Server]/src/nl/soccar/gameserver/model/Chat.java [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+        if (!optional.isPresent()) {
+            return;
+        }
+
+        optional.get().getConnection().send(new ChatMessage(-1, Privilege.ADMINISTRATOR, "You have been muted."));
+    }
+
+    public void unmutePlayer(PlayerWrapper issuer, String username) {
+        mutedNames.remove(username.toLowerCase());
+        issuer.getConnection().send(new ChatMessage(-1, Privilege.ADMINISTRATOR, String.format("You have unmuted %s.", username)));
```
Comment (Ranray) [inline Soccar [Game-Server]/src/nl/soccar/gameserver/model/Chat.java:69]: Graag een constante toevoegen voor de -1 waarde in de ChatMessage.

**Review by Ranray — CHANGES_REQUESTED** [review]

**Comment by BeeHiveJava** [comment]
Won't implement, not really necessary. Plus I'm not too proud of the format.

**Pull request closed without merging**